contracts/LockDealBundleProvider/LockDealBundleProvider.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./LockDealBundleProviderState.sol";
+import "../Provider/ProviderModifiers.sol";
+import "../ProviderInterface/IProvider.sol";
+
+contract LockDealBundleProvider is
+    ProviderModifiers,
+    LockDealBundleProviderState,
+    IProvider
+{
+    constructor(address nft, address provider) {
+        require(
+            nft != address(0x0) && provider != address(0x0),
+            "invalid address"
+        );
+        timedDealProvider = TimedDealProvider(provider);
+        lockDealNFT = LockDealNFT(nft);
+    }
+
+    ///@param providerParams[0] = leftAmount
+    ///@param providerParams[1] = startTime
+    ///@param providerParams[2] = finishTime
+    ///@param providerParams[3] = startAmount
+    function createNewPool(
+        address owner,
+        address token,
+        address[] calldata providers,
+        uint256[][] calldata providerParams
+    ) external returns (uint256 poolId) {
+        // amount for LockDealProvider = `totalAmount` - `startAmount`
+        // amount for TimedDealProvider = `startAmount`
+
+        // mint the NFT owned by the BunderDealProvider on LockDealProvider for `totalAmount` - `startAmount` token amount
+        // mint the NFT owned by the BunderDealProvider on TimedDealProvider for `startAmount` token amount
+        // mint the NFT owned by the owner on LockDealBundleProvider for `totalAmount` token transfer amount
+
+        // To optimize the token transfer (for 1 token transfer)
+        // mint the NFT owned by the BunderDealProvider with 0 token transfer amount on LockDealProvider
+        // mint the NFT owned by the BunderDealProvider with 0 token transfer amount on TimedDealProvider
+        // mint the NFT owned by the owner with `totalAmount` token transfer amount on LockDealBundleProvider
+
+        require(providers.length == providerParams.length, "providers and params length mismatch");
+
+        LockDealProvider lockDealProvider = timedDealProvider.dealProvider();
+        DealProvider dealProvider = lockDealProvider.dealProvider();
+
+        uint256 firstSubPoolId;
+        uint256 totalStartAmount;
+        for (uint256 i; i < providers.length; ++i) {
+            address provider = providers[i];
+            uint256[] memory params = providerParams[i];
+
+            // check if the provider address is valid
+            require(
+                provider == address(dealProvider) ||
+                provider == address(lockDealProvider) ||
+                provider == address(timedDealProvider),
+                "invalid provider address"
+            );
```

Comment:
if send LockDealNFT - it will fail, 
LockDealNFT doesn't have the register function 
to make sure it does not happen. we can do 2 checks:
1. if is in the approved providers
2. its not the LockDealNFT